### ISSUE TITLE
Setting nios_host_record state: absent doesn't remove the record

### ISSUE THREAD

**Issue by evilhamsterman** [infobloxopen/infoblox-ansible#206]
Using infoblox.nios_modules 1.5.0

Create a host, (not using nios_next_ip because per #63 which has been open since 2019, it is not idempotent)
```yaml
- name: Create ansible-test DNS record
  infoblox.nios_modules.nios_host_record:
    name: ansible-tests.example.com
    ipv4addrs:
      - address: 10.100.0.70
    state: present
    provider: "{{ nios_provider }}"
```
Create works successfully 

```
TASK [Create ansible-test DNS record] **********************************************
ok: [localhost]
```

Change the state to absent per docs 
```yaml
- name: Delete ansible-test DNS record
  infoblox.nios_modules.nios_host_record:
    name: ansible-tests.example.com
    ipv4addrs:
      - address: 10.100.0.70
    state: absent
    provider: "{{ nios_provider }}"
```
Task runs and does not indicate a change
```
TASK [Create ansible-test DNS record] **********************************************
ok: [localhost]
```

Check Infoblox and the record still exists

**Comment by c41m4n** [comment]
I see this issue as well on infoblox.nios_modules v1.6.1.

**Comment by dlehrman** [comment]
I was able to successfully remove the host record after excluding the ipv4 details.

e.g.
```
- name: Delete ansible-test DNS record
  infoblox.nios_modules.nios_host_record:
    name: ansible-tests.example.com
    state: absent
    provider: "{{ nios_provider }}"
```

**Comment by Torie-Coding** [comment]
> I was able to successfully remove the host record after excluding the ipv4 details.
> 
> 
> 
> e.g.
> 
> ```
> 
> - name: Delete ansible-test DNS record
> 
>   infoblox.nios_modules.nios_host_record:
> 
>     name: ansible-tests.example.com
> 
>     state: absent
> 
>     provider: "{{ nios_provider }}"
> 
> ```

How to you delete it? Even if i dont provide any ip details at all the ansible task is "ok".

**Comment by dlehrman** [comment]
> How to you delete it? Even if i dont provide any ip details at all the ansible task is "ok".

If the host record in question was created with DNS disabled (`configure_for_dns: false`, `dns: false`, or `Enable in DNS` unchecked in the GUI), you will need to include `configure_for_dns: false` in the delete task.

e.g.
```yaml
- name: Delete ansible-test DNS record
  infoblox.nios_modules.nios_host_record:
    name: ansible-tests.example.com
    configure_for_dns: false
    state: absent
    provider: "{{ nios_provider }}"
```

**Comment by Torie-Coding** [comment]
Wow... but thank you that works!

**Comment by JkhatriInfobox** [comment]
This issue is addressed in two PRs:

**PR #327** (merged) — adds `playbooks/delete_host_record.yaml` with explicit examples for:
- Deleting a DNS-enabled host record (`state: absent`, no `ipv4addrs`)
- Deleting a DNS-disabled host record (`configure_for_dns: false` + `state: absent`)
- Removing a single IP from a multi-IP record (`remove: true` + `state: absent`)

**PR `fix/nios-host-record-idempotency`** — additionally fixes the underlying `remove: true` bugs (#108) that caused WAPI errors when removing IPs, which is related to the confusion around `state: absent` + `ipv4addrs` behavior.

**TL;DR for users:** To delete an entire host record, omit `ipv4addrs` and use `state: absent`. Passing `ipv4addrs` with `state: absent` is only for removing a single IP from a multi-IP host (using `remove: true`).

Closing as addressed by PR #327.

**Comment by sarya-infoblox** [comment]
Thanks for the reports and workarounds shared here.

The correct way to delete a host record is to omit [ipv4addrs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html):
```
- name: Delete host record
  infoblox.nios_modules.nios_host_record:
    name: ansible-tests.example.com
    state: absent
    provider: "{{ nios_provider }}"
```

If the record was created with DNS disabled, also include [configure_for_dns: false](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html):
```
- name: Delete host record (DNS disabled)
  infoblox.nios_modules.nios_host_record:
    name: ansible-tests.example.com
    configure_for_dns: false
    state: absent
    provider: "{{ nios_provider }}"
```

The [ipv4addrs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) parameter with [state: absent](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is specifically for removing a single IP from a multi-IP host record (not deleting the record itself) — that requires [remove: true](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) on the address. Passing [ipv4addrs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) without [remove: true](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is a no-op by design. Closing this as the workaround is well-established and changing this behavior would risk breaking existing playbooks.

We have also updated the example playbook for the same — see [playbooks/delete_host_record.yaml](https://github.com/infobloxopen/infoblox-ansible/blob/4b495ab32ca13a60a8e18aaff763defa9175706b/playbooks/delete_host_record.yaml).
